jerry-core/ecma/builtin-objects/ecma-builtin-symbol-prototype.c

```diff
@@ -45,49 +62,44 @@
  */
 
 /**
- * The Symbol.prototype object's 'toString' routine
- *
- * See also:
- *          ECMA-262 v6, 19.4.3.2
+ * Dispatcher of the Symbol built-in's routines
  *
  * @return ecma value
  *         Returned value must be freed with ecma_free_value.
  */
-static ecma_value_t
-ecma_builtin_symbol_prototype_object_to_string (ecma_value_t this_arg) /**< this argument */
+ecma_value_t
+ecma_builtin_symbol_prototype_dispatch_routine (uint16_t builtin_routine_id, /**< built-in wide routine
+                                                                     *   identifier */
+                                                ecma_value_t this_arg, /**< 'this' argument value */
+                                                const ecma_value_t arguments_list[], /**< list of arguments
+                                                                             *   passed to routine */
+                                                ecma_length_t arguments_number) /**< length of arguments' list */
 {
-  return ecma_symbol_to_string_helper (this_arg, true);
-} /* ecma_builtin_symbol_prototype_object_to_string */
+  JERRY_UNUSED_2 (arguments_list, arguments_number);
 
-/**
- * The Symbol.prototype object's 'valueOf' routine
- *
- * See also:
- *          ECMA-262 v6, 19.4.3.3
- *
- * @return ecma value
- *         Returned value must be freed with ecma_free_value.
- */
-static ecma_value_t
-ecma_builtin_symbol_prototype_object_value_of (ecma_value_t this_arg) /**< this argument */
-{
-  return ecma_symbol_to_string_helper (this_arg, false);
-} /* ecma_builtin_symbol_prototype_object_value_of */
+  ecma_value_t sym = ecma_symbol_this_value (this_arg);
 
-/**
- * The Symbol.prototype object's '@@toPrimitive' routine
- *
- * See also:
- *          ECMA-262 v6, 19.4.3.3
- *
- * @return ecma value
- *         Returned value must be freed with ecma_free_value.
- */
-static ecma_value_t
-ecma_builtin_symbol_prototype_object_to_primitive (ecma_value_t this_arg) /**< this argument */
-{
-  return ecma_builtin_symbol_prototype_object_value_of (this_arg);
-} /* ecma_builtin_symbol_prototype_object_to_primitive */
+  if (builtin_routine_id < ECMA_SYMBOL_PROTOTYPE_TO_STRING)
+  {
+    return ecma_copy_value (sym);
+  }
+
+  if (ECMA_IS_VALUE_ERROR (sym))
```

Comment:
This check should be before `if (builtin_routine_id < ECMA_SYMBOL_PROTOTYPE_TO_STRING)`